### DISCUSSION TITLE
MQTT seems doesn't work

### DISCUSSION THREAD

**Discussion by 4sp3c** [Koenkk/zigbee2mqtt#10037] [General]
Good morning everyone. I just installed sonoff coordinator and I have 2 sonoff probe. I flashed the coordinator with the last firmware and I associated a probe correctly (I can see that on configuration.yalm) 
I installed mosquitto and I running on my pc. I have this configuration

for zigbee:
homeassistant: false
permit_join: false
mqtt:
  base_topic: zigbee2mqtt
  server: mqtt://localhost:1883
serial:
  port: /dev/ttyUSB1
advanced:
  homeassistant_legacy_entity_attributes: false
  legacy_api: false
  log_level: debug
device_options:
  legacy: false

for MQTT
persistence_location /var/lib/mosquitto/
log_dest file /var/log/mosquitto/mosquitto.log
listener 1883
allow_anonymous true

I started with sudo privileges zigbee2mqtt with
sudo npm start

the system started and I had this log

Zigbee2MQTT:debug 2021-12-06 10:22:45: Loaded state from file /opt/zigbee2mqtt/data/state.json
Zigbee2MQTT:info  2021-12-06 10:22:45: Logging to console and directory: '/opt/zigbee2mqtt/data/log/2021-12-06.10-22-45' filename: log.txt
Zigbee2MQTT:debug 2021-12-06 10:22:45: Removing old log directory '/opt/zigbee2mqtt/data/log/2021-12-06.08-59-24'
Zigbee2MQTT:info  2021-12-06 10:22:45: Starting Zigbee2MQTT version 1.22.1 (commit #d95d6cdc)
Zigbee2MQTT:info  2021-12-06 10:22:45: Starting zigbee-herdsman (0.13.176)
Zigbee2MQTT:debug 2021-12-06 10:22:45: Using zigbee-herdsman with settings: '{"adapter":{"concurrent":null,"delay":null,"disableLED":false},"backupPath":"/opt/zigbee2mqtt/data/coordinator_backup.json","databaseBackupPath":"/opt/zigbee2mqtt/data/database.db.backup","databasePath":"/opt/zigbee2mqtt/data/database.db","network":{"channelList":[11],"extendedPanID":[221,221,221,221,221,221,221,221],"networkKey":"HIDDEN","panID":6754},"serialPort":{"path":"/dev/ttyUSB1"}}'
Zigbee2MQTT:info  2021-12-06 10:22:47: zigbee-herdsman started (resumed)
Zigbee2MQTT:info  2021-12-06 10:22:47: Coordinator firmware version: '{"meta":{"maintrel":1,"majorrel":2,"minorrel":7,"product":1,"revision":20210708,"transportrev":2},"type":"zStack3x0"}'
Zigbee2MQTT:debug 2021-12-06 10:22:47: Zigbee network parameters: {"channel":11,"extendedPanID":"0x00124b00258d48c5","panID":6754}
Zigbee2MQTT:info  2021-12-06 10:22:47: Currently 0 devices are joined:
Zigbee2MQTT:info  2021-12-06 10:22:47: Zigbee: disabling joining new devices.
Zigbee2MQTT:info  2021-12-06 10:22:47: Connecting to MQTT server at mqtt://localhost:1883
Zigbee2MQTT:debug 2021-12-06 10:22:47: Using MQTT anonymous login

and stop. The logs finished here and MQTT is not connected. How can I solve it? 
Any helps will be appreciated.
Thank you to everyone

**Comment by rommelfreddy** [comment]
@4sp3c did you figured out the issue?
i have the same issue, but i have no idea what could be wrong. 

i have a similar setup like you.

**Reply by 4sp3c** [reply]
Hi @rommelfreddy, if I remember well, I changed only localhost with 127.0.0.1
try and give me a feedback

**Reply by rommelfreddy** [reply]
i figured out, that the protocol type "websockets" will produce the error, that the port binding of mqtt will only for the ipv6 address. 
After changing it to "mqtt" the problem is resolved and the port got also binded for ipv4.

Got the "websockets" value from two tutorials. 
I am honest - i have no idea where the difference is. 
Never mind - now it works like expected.
